src/Tools/dotnet-counters/CounterMonitor.cs

```diff
@@ -291,6 +323,34 @@ private bool BuildDiagnosticsClient()
                     return false;
                 }
             }
+            else if (port.Length != 0)
+            {
+                ReversedDiagnosticsServer server = new ReversedDiagnosticsServer(port);
+                server.Start();
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                {
+                    Console.WriteLine($"Waiting for connection on {port}");
+                    Console.WriteLine($"Start an application with the following environment variable: DOTNET_DiagnosticPorts={port},suspend");
+                }
+                else
+                {
+                    Console.WriteLine($"Waiting for connection on {Path.GetFullPath(port)}");
+                    Console.WriteLine($"Start an application with the following environment variable: DOTNET_DiagnosticPorts={Path.GetFullPath(port)},suspend");
+                }
+
+                ManualResetEvent shouldExit = new ManualResetEvent(false);
+                _ct.Register(() => shouldExit.Set());
+
+                Task acceptTask = Task.Run(() =>
+                {
+                    IpcEndpointInfo endpointInfo = server.Accept(TimeSpan.FromMilliseconds(Int32.MaxValue));
+                    Console.WriteLine($"Connection established");
+                    _diagnosticsClient = new DiagnosticsClient(endpointInfo.Endpoint);
+                    _diagnosticsClient.ResumeRuntime();
```

Comment:
If we immeidately resume the runtime, we might lose information, e.g., Assembly loading events.  We should start the requested sessions before resuming.